src/Application/Common/Interfaces/IMailService.cs

```diff
@@ -0,0 +1,12 @@
+using System.Threading.Tasks;
+using Codidact.Authentication.Domain.Entities;
+
+
+namespace Codidact.Authentication.Application.Common.Interfaces
+{
+    public interface IMailService
+    {
+        Task SendEmailAsync(ApplicationUser user, string subject, string message);
```

Comment:
This should not be in the interface, it should be a private method of `MailService`.